packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -161,6 +170,38 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
   const headerProps = useAdaptedSelector(getHeaderProps);
   const errorBarProps = usePropsFor(ErrorBar);
 
+  /* @conditional-compile-remove(image-overlay) */
+  useEffect(() => {
+    if (overlayImageItem === undefined) {
+      return;
+    }
+    const messages = messageThreadProps.messages.filter((message) => {
+      return message.messageId === overlayImageItem?.messageId;
+    });
+    if (messages.length <= 0 || messages[0].messageType !== 'chat') {
+      return;
+    }
+    const message = messages[0] as ChatMessage;
+    if (overlayImageItem.imageSrc === '' && message.inlineImages && message.inlineImages?.length > 0) {
+      const inlineImages = message.inlineImages.filter((attachment) => {
+        return attachment.id === overlayImageItem?.attachmentId;
+      });
+      if (
+        inlineImages.length <= 0 ||
+        inlineImages[0].fullSizeImageSrc === undefined ||
+        inlineImages[0].fullSizeImageSrc === '' ||
+        overlayImageItem.imageSrc === inlineImages[0].fullSizeImageSrc
+      ) {
+        return;
+      }
+      setOverlayImageItem({
+        ...overlayImageItem,
+        imageSrc: inlineImages[0].fullSizeImageSrc
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Do we need to disable this lint?

Reply:
> Do we need to disable this lint?

Lint is disabled on this line because we don't want to listen to the change of overlayImageItem. Because we're setting the overlayImageItem, listen to overlayImageItem and set it will create a infinite loop.

Reply:
Sounds good! Can we add a comment with this that describes that?